pkg/konnector/konnector_reconcile.go

```diff
@@ -120,6 +120,7 @@ func (r *reconciler) reconcile(ctx context.Context, binding *kubebindv1alpha1.AP
 	ctrl, err := r.newClusterController(
 		binding.Spec.KubeconfigSecretRef.Namespace+"/"+binding.Spec.KubeconfigSecretRef.Name,
 		providerNamespace,
+		binding.Name,
```

Comment:
it's a little bit more complicated. Compare the `ctrlContext` logic above. There The controller context contains a list of bindings it is started for. So it is also responsible for all its bindings, i.e. not only `binding.Name` here.